src/main/java/Duke.java

```diff
@@ -1,10 +1,179 @@
+                        throw new DukeException("____________________________________________________________\n" +
+                                " OOPS! Your Only Friend cannot take in an empty " + command[0] +
+                                "\n Make sure " + command[0] + "has a description!\n" +
+                                "____________________________________________________________\n");
+                    }
+                    if (command[0].equals("todo")) {
+                        task = new ToDo(command[1]);
+                    } else if (command[0].equals("deadline")) {
+                        task = new Deadline(command[1]);
+                    } else if (command[0].equals("event")) {
+                        task = new Event(command[1]);
+                    } else {
+                        throw new DukeException("____________________________________________________________\n" +
+                                " OOPS! Turns out Your Only Friend does not know what that is :(\n" +
+                                "____________________________________________________________\n");
+
+                    }
+
+                    listOfTasks.add(task);
+                    System.out.println("____________________________________________________________\n"
+                            + " Got it. I've added this task:\n"
+                            + task.toString() + "\n"
+                            + " Now you have " + listOfTasks.size() + (listOfTasks.size() <= 1 ? " task in the list." : " tasks in the list.")
+                            + "\n____________________________________________________________\n");
+                }
+
+                whatToDoCall = whatToDo.nextLine().trim().toLowerCase();
+                command = whatToDoCall.split(" ", 2);
+            }
+
+            try {
+                File directory = new File("./data/");
+
+                if (!directory.exists()) {
+                    directory.mkdirs();
+                }
+
+                saveToFile("./data/duke.txt", stringToSave(listOfTasks));
+            } catch (IOException e) {
+                System.out.println("Error while saving, please try again!");
+            }
+        } catch (DukeException exception) {
+            System.out.println(exception.getMessage());
+        }
+
+        System.out.println("____________________________________________________________\n"
+                + " Bye. Hope to see you again soon!\n - Your Only Friend\n"
+                + "____________________________________________________________\n");
+
+
+    }
+
+    public static void load(ArrayList<Task> listOfTasks) {
+
+        try {
+            File file = new File("./data/duke.txt");
+            Scanner whatToDo = new Scanner(file);
+
+            while (whatToDo.hasNext()) {
+                String readText = whatToDo.nextLine();
+                char taskType = readText.charAt(1);
+                char doneOrNot = readText.charAt(5);
```

Comment:
this variable seems to serve the purpose of a boolean so maybe you could name it as "isDone".